Feat: Added the new feature stay signed in

danieltate888 opened the pull request:
## What happens here?
Added the new "Stay Signed In" feature with the following changes:
- Added `token.service.ts` to parse 'id' from the access token and inject it into the request object.
- Added `jwt-refresh-auth.guard.ts` to validate the refresh token.
- Added `jwt-access-auth.guard.ts` to validate the access token.
- Created `combined-auth.guard.ts` to combine the two guards and `token.service.ts`, adjusting access token and refresh token validity.
- Implemented refresh token and access token JWT strategies to handle validation logic.
- Added the `logout` mutation in `auth.service.ts` to clear the refresh token from HTTP-only cookies and the database.
- Modified token generation logic in `auth.service.ts`.
- Updated Jest test files related to the changes.
- Added `combined-auth.guard.spec.ts` to test the combined auth guard.
- Imported an error filter into `app.module.ts`.
- Added `AuthModule` into `ExchangeKeyModule` and `UserModule`.
- Changed the `GqlAuthGuard` to `CombinedAuthGuard` in `ExchangeKeyResolver` and `UserExchangeResolver`.
- Modified the `generateAccessToken` and `revokeTokens` methods in `auth.service.ts`.
- Added the `generateAccessToken` method into `auth.resolver.ts`.
- Exported `EmailVerificationService` at `auth.module.ts`.
- Modified `combined-auth.guard.spec.ts`.
- Changed error handling from `UnauthorizedException` to `CustomException` to add response codes.
- Deleted `GqlAuthGuard`.
- Added `GqlHttpExceptionFilter` and `CustomException`.
- Added `ACCESS_TOKENS_NOT_SAME` and `BOTH_TOKENS_INVALID` into `code.ts`.

## How do I know this is working?
1. Click Stay signed in at login page
2. Changed the access token and refresh token validation at backend in the auth.service.ts(login and generateAccessToken method)
3. Request any one method that will pass CombinedAuthGuard to see the effect

## Screenshot:
Jest:
<img width="1080" alt="image" src="https://github.com/user-attachments/assets/e92e0a7a-b9b8-4e60-979c-2abbdd164345">

Postman:
refresh token and access token are both invalid:
<img width="1526" alt="image" src="https://github.com/user-attachments/assets/2732c75f-114e-4b27-8f06-3d9495aedb25">

Database:
refresh token added to database:
<img width="1080" alt="image" src="https://github.com/user-attachments/assets/e2ccbdc8-c89d-4969-9abc-f70468bbe6ec">

## Any notes?
Resolver: CP-20

Chris-Wang commented on src/modules/auth/auth.resolver.ts:
```diff
@@ -10,15 +10,22 @@ import { passwordUpdateSchema } from '@/validation/schemas/auth/password.update'
+    @Args('staySignedIn') staySignedIn: boolean,
+    @Context() context: any,
+  ): Promise<Result> {
+    console.log('开始登录');
```
remove console log

Chris-Wang commented on src/modules/auth/auth.service.ts:
```diff
@@ -17,29 +17,46 @@ import {
+  async login(
+    email: string,
+    password: string,
+    staySignedIn: boolean,
```
change name to isStaySignedIn

Chris-Wang commented on src/modules/user/user.service.ts:
```diff
@@ -28,10 +31,10 @@ export class UserService {
   async update(id: string, entity: DeepPartial<User>): Promise<boolean> {
     const res = await this.userRepository.update(id, entity);
-    console.log(res);
+    // console.log(res);
```
we can remove this

Chris-Wang commented on package.json:
```diff
@@ -35,6 +35,7 @@
     "bcryptjs": "^2.4.3",
     "ccxt": "^4.1.48",
     "class-validator": "^0.14.0",
+    "cookie-parser": "^1.4.6",
```
is cookie-parser still being used?

Chris-Wang reviewed (changes requested):
1. Resolve conflict
2. Add unit test of stay signed path

Chris-Wang commented on src/modules/auth/guards/combined-auth.guard.spec.ts:
```diff
@@ -0,0 +1,158 @@
+
+      (refreshJwtAuthGuard.canActivate as jest.Mock).mockImplementation(async () => {
+        const args = context.getArgs();
+        args[2].req.user = { id: 1 }; // 确保 req 对象存在并设置模拟用户
```
remove comment, or write comment in English

Chris-Wang commented on src/modules/auth/guards/combined-auth.guard.spec.ts:
```diff
@@ -0,0 +1,158 @@
+    getContext: jest.fn().mockReturnValue(gqlContext),
+  } as unknown as ExecutionContext;
+
+  // 模拟 GqlExecutionContext.create 方法
```
remove comment, or write comment in English

Commit:
Feat: Stay Signed In

Added the new "Stay Signed In" feature with the following changes:
- Added `token.service.ts` to parse 'id' from the access token and inject it into the request object.
- Added `jwt-refresh-auth.guard.ts` to validate the refresh token.
- Added `jwt-access-auth.guard.ts` to validate the access token.
- Created `combined-auth.guard.ts` to combine the two guards and `token.service.ts`, adjusting access token and refresh token validity.
- Implemented refresh token and access token JWT strategies to handle validation logic.
- Added the `logout` mutation in `auth.service.ts` to clear the refresh token from HTTP-only cookies and the database.
- Modified token generation logic in `auth.service.ts`.
- Updated Jest test files related to the changes.
- Added `combined-auth.guard.spec.ts` to test the combined auth guard.
- Imported an error filter into `app.module.ts`.
- Added `AuthModule` into `ExchangeKeyModule` and `UserModule`.
- Changed the `GqlAuthGuard` to `CombinedAuthGuard` in `ExchangeKeyResolver` and `UserExchangeResolver`.
- Modified the `generateAccessToken` and `revokeTokens` methods in `auth.service.ts`.
- Added the `generateAccessToken` method into `auth.resolver.ts`.
- Exported `EmailVerificationService` at `auth.module.ts`.
- Modified `combined-auth.guard.spec.ts`.
- Changed error handling from `UnauthorizedException` to `CustomException` to add response codes.
- Deleted `GqlAuthGuard`.
- Added `GqlHttpExceptionFilter` and `CustomException`.
- Added `ACCESS_TOKENS_NOT_SAME` and `BOTH_TOKENS_INVALID` into `code.ts`.

Resolve: CP-20

Chris-Wang reviewed (approved):
LGTM
